applications/GeoMechanicsApplication/tests/cpp_tests/stub_constitutive_law.h

```diff
@@ -13,14 +13,24 @@
 
 #include "includes/constitutive_law.h"
 
+#include <gmock/gmock.h>
+
 namespace Kratos::Testing
 {
 
+ConstitutiveLaw::Pointer MakeStubConstitutiveLaw();
+
 class StubConstitutiveLaw : public ConstitutiveLaw
 {
 public:
-    bool                                   RequiresInitializeMaterialResponse() override;
-    [[nodiscard]] ConstitutiveLaw::Pointer Clone() const override;
+    StubConstitutiveLaw()
+    {
+        ON_CALL(*this, RequiresInitializeMaterialResponse).WillByDefault(testing::Return(false));
+        ON_CALL(*this, Clone).WillByDefault([]() { return MakeStubConstitutiveLaw(); });
```

Comment:
I'd propose to simplify this, but I'd only do that if there's something else that also needs to change (not worth waiting on the pipelines once more in my opinion)
```suggestion
        ON_CALL(*this, Clone).WillByDefault([]() { return std::make_shared<StubConstitutiveLaw> });
```

Reply:
Hmm... that's odd. I've tried exactly this before and then it didn't work. I thought it had to do with attempting to construct an instance of `StubConstitutiveLaw` from its constructor. But now that I've tried it again (to see which error message I got exactly) it all works as expected. As a result, we no longer need the implementation file (`.cpp`), since it had become empty. :-)